app/one_click_unsubscribe/rest.py

```diff
@@ -52,3 +60,57 @@ def get_unsubscribe_request_data(notification, email_address):
         "service_id": notification.service_id,
         "email_address": email_address,
     }
+
+
+def create_unsubscribe_request_reports_summary(service_id):
+    reports_summary = {"batched_reports_summaries": [], "unbatched_report_summary": {}}
+    existing_unsubscribe_reports_summaries = []
+    # Check for existing unsubscribe reports and create their summaries
+    if unsubscribe_request_reports := get_unsubscribe_request_reports_dao(service_id):
+        existing_unsubscribe_reports_summaries = _create_existing_unsubscribe_request_reports_summary(
+            unsubscribe_request_reports,
+        )
+        reports_summary["batched_reports_summaries"] = existing_unsubscribe_reports_summaries
+    # Check for unsubscribe requests and create a summary for them
+    if unbatched_unsubscribe_requests := get_unbatched_unsubscribe_requests_dao(service_id):
+        if existing_unsubscribe_reports_summaries:
+            earliest_timestamp = existing_unsubscribe_reports_summaries[0]["latest_timestamp"] + timedelta(seconds=3)
```

Comment:
Why are we adding 3 seconds here? Feels like we might miss some requests in the gap

Reply:
That was a mistake, It was my intention to use 3 microseconds. I had wanted some kind of demarcation between an `unsubscribe_request_report` `earliest_timestamp` and the previous `unsubscribe_request_report` `latest_timestamp`.
I have removed it.